### ISSUE TITLE
add support to select the type of indentation

### ISSUE THREAD

**Issue by cjkumaresh** [sahat/megaboilerplate#129]
It would be great if we have the option to select the type of indentation(2 spaces or 4 spaces).By default it is 2 spaces. But 4 spaces indentation is also widely used i believe

**Comment by sahat** [comment]
Hi @cjkumaresh, this is easy enough to do with code blocks that get replaced, but more difficult with existing files.

Take `server.js` for instance. It's a file indented with 2 spaces and it contains a bunch of `//=` comment lines where additional code is inserted. I can control indent level of inserted code, but not existing code in `server.js`. At least I don't know of any tool that can intelligently take all code (of any file extension) and indent it by specified amount. Many IDEs do this already but I don't know of a NPM module that can indent any arbitrary block of text. And furthermore because each file extension has its own indentation rules, for example HTML vs JS vs CSS, it would be very difficult to implement this from scratch.

**To sum it up**: it may be easier for you to press one keyboard shortcut in Webstorm/Sublime Text/Atom to re-indent a file according to your preferences than implement it at Mega Boilerplate level.

**Comment by cjkumaresh** [comment]
Yeah that make sense!
